packages/eslint-plugin-orbit-components/src/rules/arrowFunctionsCount.ts

```diff
@@ -0,0 +1,48 @@
+import * as t from "@babel/types";
+import type { Rule } from "eslint";
+
+const LIMIT = 5;
+export const ERROR =
+  "You are using too many arrow function expressions in one single styled component. This can negatively impact performance. Consider using one single function to desctructure all props and return one single css helper function if possible.";
```

Comment:
Nice! I would phrase it a little differently to sound less accusatory: 
```suggestion
  "Using many arrow function expressions in a single styled component can negatively impact performance. Consider using one single function to destructure all props and return one single css helper function if possible.";
```